api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -516,8 +519,15 @@ private void handleExpiredCredits(
       vwbUserService.unlinkBillingAccountForUserPod(user);
     }
 
-    userInitialCreditsExpiration.setExpirationCleanupTime(clockNow());
-    userDao.save(user);
+    try {
+      mailService.alertUserInitialCreditsExpired(user);
+      userInitialCreditsExpiration.setExpirationCleanupTime(clockNow());
+      userDao.save(user);
+    } catch (MessagingException e) {
+      logger.error(
+          "Failed to send initial credits expired warning notification for user {}",
+          user.getUserId());
+    }
   }
 
   private void handleExpiringSoonCredits(
```

Comment:
I can't leave a comment in exactly the right place, but we need to update when this is called from `checkExpiration`. This will only be called if `getApproachingExpirationNotificationTime` is null, so we'll only send an email to the user once on the 90 day mark. There will need to be some logic to ensure the second email is only send once the second time threshold has been crossed.

Reply:
Did you handle this scenario? I might be missing it, but I'm seeing that `checkExpiration` still makes the same `null == userInitialCreditsExpiration.getApproachingExpirationNotificationTime()` check which will only be true once. After the first `expiringSoon` email is sent, that check will evaluate as false and we won't send additional emails

Reply:
That will ensure that `areCreditsExpiringSoon` still returns true, but it doesn't change the fact that `null == userInitialCreditsExpiration.getApproachingExpirationNotificationTime()` will always evaluate to false after the first email is sent out.